pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -203,6 +203,10 @@ type CiliumBGPNeighbor struct {
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=4294967295
 	PeerASN int64 `json:"peerASN"`
+	// AuthSecret is the name of the secret to use to fetch a TCP authentication
+	// password for this peer.
+	// +kubebuilder:validation:Optional
+	AuthSecretRef *string `json:"authSecretRef"`
```

Comment:
Consider adding the `omitempty` keyword to omit the field from JSON serialization when unset.

Reply:
I think hard-coding `kube-system` in the spec like this is problematic, it would actually be the namespace the helm chart is deployed in?

The user may also in that case need a way to configure what namespaces they want RBAC to allow to read from, the nice part about using a single namespace as currently is the permissions are easy to understand and there's no risk of allowing too many permissions.

Reply:
I've added `omitempty`.

Reply:
IMO, it is a bit overkill for this use case. We just want to pass the password here, and the Secret object is stable and pretty primitive (it's effectively an arbitrary KV). So, I can't come up with any scenario that we need to specify another object. Even if users are managing passwords in a way other than Secret (e.g., SealedSecret or Hashicorp Vault), they usually support generating Secret, so we still don't have to change Group/Kind.

For the namespace, I don't have a strong opinion about how to configure it. In both ways, users need to specify a namespace that the Secret exists and RBAC to allow access.

Currently, it is simple because the namespace is known at installation time, so we can generate the RBAC setting with Helm with a single configuration knob, `secretsNamespace`.

On the other hand, in the way users specify it on CRD, we cannot know the namespace at installation time, so users need to write RBAC manually. However, it gives users flexibility.

Personally, I think we're okay with the current form for the above reason. WDYT?

Reply:
I think maybe it would make sense to go for the middle ground of adding `AuthReference`, but *only* putting a `Name` field inside it for now. If later extensions are wanted, then it would be possible to add something like the fields outlined above, without breaking compat.

Reply:
+1 on ^ @dgl

Reply:
@dgl thanks for adding `omitempty`. Are you still planning to implement https://github.com/cilium/cilium/pull/23759#discussion_r1343613666? Either way, you can consider this conversation resolved since you added `omitempty` and others are fine with the current approach for exposing the auth secret reference.

Reply:
I'm going to leave it. when I thought about it actually if it was wanted later we could add a `authRef` field instead, so I don't think there's a need to have an extra struct just to potentially keep compatibility later.